backend/api/src/main/kotlin/software/shonk/adapters/incoming/ShorkInterpreterController.kt

```diff
@@ -216,6 +216,26 @@ fun Route.configureShorkInterpreterControllerV1() {
         val errors = shorkUseCase.getCompilationErrors(compileErrorsRequest.code)
         call.respond(mapOf("errors" to errors))
     }
+
+    get("/lobby/{lobbyId}/settings") {
+        val lobbyId =
+            call.parameters["lobbyId"]?.toLongOrNull()
+                ?: return@get call.respond(HttpStatusCode.BadRequest)
+
+        val settingsResult = shorkUseCase.getLobbySettings(lobbyId)
+
+        settingsResult.onFailure {
+            call.respond(HttpStatusCode.BadRequest, it.message ?: UNKNOWN_ERROR_MESSAGE)
```

Comment:
If there was some unknown error that is not clearly attributed to the request by the client the response should not be Bad Request. Internal Server Error would be more appropriate.

Reply:
I replaced the BadRequest with InternalServerError. But I'm still not sure if you want it replaced or added alongside BadRequest to have 2 cases